src/app/static/src/app/store/modelOptions/mutations.ts

```diff
@@ -81,6 +78,56 @@ export const mutations: MutationTree<ModelOptionsState> = {
     },
 
     [ModelOptionsMutation.SetModelOptionsVersion](state: ModelOptionsState, action: PayloadWithType<VersionInfo>) {
-        state.version = action.payload
+        state.version = action.payload;
     }
 };
+
+function checkOptionsValid(formMeta: DynamicFormMeta): boolean {
+    let valid = true;
+    formMeta.controlSections.forEach(section => {
+        section.controlGroups.forEach(group => {
+            group.controls.forEach(control => {
+                // We could go further and check that if it is null or empty that this isn't
+                // a required control but just assuming this for now seems ok
+                if (control.value != null && control.value != "" && hasOptions(control)) {
+                    valid = valid && checkControlOptionValid(control);
```

Comment:
if you find something that sets valid to false, would you not want to break out of the loops? for this scenario you could replace `.forEach` with `.every` and that will break out of the for loop as soon as you return false

https://stackoverflow.com/questions/34425795/breaking-out-of-an-inner-foreach-loop

just a suggestion though, happy for you to not do it since the forms arent huge

Reply:
Thanks, that's nice!